src/py/flwr/common/message.py

```diff
@@ -182,14 +226,67 @@ def metadata(self) -> Metadata:
     @property
     def content(self) -> RecordSet:
         """The content of this message."""
+        if self._content is None:
+            raise ValueError(
+                "Message content is None. Use <message>.has_content() "
+                "to check if a message has content."
+            )
         return self._content
 
     @content.setter
     def content(self, value: RecordSet) -> None:
         """Set content."""
-        self._content = value
+        if self._error is None:
+            self._content = value
+        else:
+            raise ValueError("A message with an error set cannot have content.")
+
+    @property
+    def error(self) -> Error:
+        """Error captured by this message."""
+        if self._error is None:
+            raise ValueError(
+                "Message error is None. Use <message>.has_error() "
+                "to check first if a message carries an error."
+            )
+        return self._error
+
+    @error.setter
+    def error(self, value: Error) -> None:
+        """Set error."""
+        if self.has_content():
+            raise ValueError("A message with content set cannot carry an error.")
+        self._error = value
+
+    def has_content(self) -> bool:
+        """Return True if message has content, else False."""
+        return self._content is not None
+
+    def has_error(self) -> bool:
+        """Return True if message has an error, else False."""
+        return self._error is not None
+
+    def create_error(
+        self,
+        error: Error,
+        ttl: str,
+    ) -> Message:
+        """Construct valid response message indicating an error happened.
```

Comment:
```suggestion
        """Construct valid reply message indicating an error happened.
```

Would you mind changing "response" to "reply"? Both words are valid, actually. But it would be better if we can use the same words, since the "reply message" is widely used in `create_reply` and in `Driver`.